### PR TITLE
Fix #186: Document uriBaseId

### LINKED ISSUE #186: Document uriBaseId
Document it in general, and specifically on:
- `file`
- `fileChange`
- `physicalLocation`
- `stackFrame`

### PR DESCRIPTION
**Pull request by ghost**
Document it in general, and also specifically on:
- `file`
- `fileChange`
- `physicalLocation`
- `stackFrame`

Also:
- Reword "where the tool produced the result" to "where the tool detected the result" in a few places.

@michaelcfanning

### CODE REVIEW

File: document/CoreFormat.mdk [hunk 1]
```diff
@@ -57,6 +57,83 @@ nor are hyphen, underscore, or tilde.
...
+By a convention agreed upon between the analysis tool and the log file consumers,
+this indicates that the relative URI is expressed relative to the root of the source tree
+in which the file appears.
+
+```
+"results": [
+  {
+    "resultFile": {
+      "uri": "drivers/video/hidef/driver.c",
+      "uriBaseId": "src"
```
Comment (michaelcfanning) [inline document/CoreFormat.mdk]: Can I suggest that we populate uriBaseId with examples that correspond to common patterns on windows/unix for specifying variables. We could use %src_root% for one, for examples, and $bindrop for the other. VS uses patterns such as $(SolutionDir), Android Studio uses constructs such as $PROJECT_DIR$. I think the example will be more illustrative if the example itself resembles something that is intended to be expanded. 'src' looks like a path fragment.

File: document/CoreFormat.mdk [hunk 2]
```diff
@@ -57,6 +57,83 @@ nor are hyphen, underscore, or tilde.
...
+    }
+  }
+]
+```
+~
+
+~ IsoNote
+There are various reasons for providing URI base id properties:
+
+1. Portability: A log file that contains relative URIs together with  URI base id properties
```
Comment (michaelcfanning) [inline document/CoreFormat.mdk]: Related, another important property is determinism of output, to drive caching/scheduling decisions when examining output from diverse machines. We will want to explicitly address how to ensure determinism in SARIF output.

**Comment by michaelcfanning** [comment]
+1. Nice update. See my minor suggestion around the uriBaseId examples.

File: document/CoreFormat.mdk (hunk 2 above)
Comment (ghost) [inline document/CoreFormat.mdk]: Oh, right, I'll mention that as well.

File: document/CoreFormat.mdk (hunk 1 above)
Comment (ghost) [inline document/CoreFormat.mdk]: We can certainly talk about it. This was a considered choice. The thing is, these values aren't intended to be "expanded" so much as "looked up". Those substitution syntaxes make sense in the context of a larger string (%SDXROOT%\foo\bar.cs), where you have to be able to tell that the funny-looking thing isn't part of the path. Here, we've cleverly separated the "key" out from the larger string and so (as the text above says), it "need not conform to any particular syntax or follow any particular naming convention".

Thoughts?

Comment (ghost) [inline document/CoreFormat.mdk]: The thing is that giving the example you provided suggests that the value of `baseUriId` has to _be_ something like an environment variable or an MSBuild variable, and it doesn't, which is why I wrote the examples as I did.

File: document/CoreFormat.mdk (hunk 2 above)
Comment (ghost) [inline document/CoreFormat.mdk]: ... and I'll file an issue to write an Annex on determinism.

**Pull request merged**